lib/utils/report.mjs

```diff
@@ -115,6 +125,45 @@ export default function report(problem) {
 	result.warn(warningMessage, warningProperties);
 }
 
+/**
+ * @param {Problem} problem
+ */
+function problemRangeDeprecations(problem) {
+	if (problem.result.stylelint.quietDeprecationWarnings) return;
+
+	if (!problem.node) {
+		emitDeprecationWarning(
+			`Omitting the \`node\` argument in the \`utils.report()\` function is deprecated ("${problem.ruleName}").`,
+			'REPORT_AMBIGUOUS_POSITION',
+			`Please pass a \`node\` argument in the \`utils.report()\` function of "${problem.ruleName}".`,
+		);
```

Comment:
suggestion
minor

We already had `report` throwing when `node` wasn't passed.
I don't think this `emitDeprecationWarning` is necessary.
It should remain a simple `throw`.

Reply:
The current code throws when neither `node` or `line` are provided.
According to that logic it would be acceptable to pass only `node` or only `line`.
Rule authors might today omit `node` and provide `line`.

We need to communicate that `node` is now required and that `line` is deprecated.

This shape of the code also aligns with the intended form for the next major: https://github.com/stylelint/stylelint/blob/b0525066e26cd5a1f45a2a746be945c8150046df/lib/utils/report.cjs#L99-L103

In the next major we do:
- remove the existing `throw`
- `throw` instead of warn about deprecations

So I intentionally added this throw when `node` is missing as this is both the most informative and the most direct path forwards.